pkg/controller/build/build_controller.go

```diff
@@ -0,0 +1,1332 @@
+		}
+
+		// Now it's a machineconfig
+		mc, err := ctrlcommon.MachineConfigFromIgnConfig(pool.Name, entitledConfigName, ignConfig)
+		if err != nil {
+			glog.Warningf("Could not create machineconfig for entitlements: %s", err)
+		}
+
+		// Add it to the list for this pool
+		_, err = ctrl.client.MachineconfigurationV1().MachineConfigs().Create(context.TODO(), mc, metav1.CreateOptions{})
+		if err != nil {
+			glog.Warningf("Failed to add entitlements to layered pool: %s", err)
+		}
+	}
+
+}
+
+// annotatePoolWithNewestImage looks in the corresponding image stream for a pool and annotates the name of the image, and it's
+// corresponding rendered-config, which it retrieves from the image's docker metadata labels that we added during our build
+func (ctrl *Controller) annotatePoolWithNewestImage(imageStream *imagev1.ImageStream, pool *mcfgv1.MachineConfigPool) error {
+
+	// We don't want to crash if these are empty
+	if pool.Annotations == nil {
+		pool.Annotations = map[string]string{}
+	}
+
+	// Grab the latest tag from the imagestream. If we don't have one, nothing happens
+	for _, tag := range imageStream.Status.Tags {
+		if len(tag.Items) == 0 {
+			continue
+		}
+
+		// I might have an older image that has right machine config content, but some
+		// other content might have changed (like, I dunno, base image) so we shouldn't go back
+		// to older images
+		image := tag.Items[0]
+
+		// If this is different than our current tag, grab it and annotate the pool
+		glog.Infof("imagestream %s newest is: %s (%s)", imageStream.Name, image.DockerImageReference, image.Image)
+		if pool.Spec.Configuration.Name == image.Image {
+			// We're already theer, don't touch it
+			return nil
+		}
+
+		// get the actual image so we can read its labels
+		fullImage, err := ctrl.imageclient.ImageV1().Images().Get(context.TODO(), image.Image, metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("Could not retrieve image %s: %w", image.Image, err)
+		}
+
+		// We need the labels out of the docker image but it's a raw extension
+		dockerLabels := struct {
+			Config struct {
+				Labels map[string]string `json:"Labels"`
+			} `json:"Config"`
+		}{}
+
+		// Get the labels out and see what config this is
+		err = json.Unmarshal(fullImage.DockerImageMetadata.Raw, &dockerLabels)
+		if err != nil {
+			return fmt.Errorf("Could not get labels from docker image metadata: %w", err)
+		}
+
+		// Tag what config this came from so we know it's the right image
+		if machineconfig, ok := dockerLabels.Config.Labels["machineconfig"]; ok {
+			pool.Annotations[ctrlcommon.ExperimentalNewestLayeredImageEquivalentConfigAnnotationKey] = machineconfig
+			pool.Spec.Configuration.ObjectReference = corev1.ObjectReference{
+				Kind: "Image",
+				Name: image.Image,
+			}
+			// TODO(jkyros): Kind of cheating using this as metadata showback for the user until we figure out our "level triggering" strategy
+			pool.Spec.Configuration.Source = []corev1.ObjectReference{
```

Comment:
Talking with Colin still, we were talking about how nice it would be to have some sort of history on the pool -- something like "this update changed X" and being able to step back through those.